Update the DejaVu fonts and add a package ‘PTerm-Fonts’ embedding DejaVu Sans Mono

Rinzwind opened the pull request:
This pull request updates the DejaVu fonts to v2.37 and adds a package ‘PTerm-Fonts’ embedding DejaVu Sans Mono. In the baseline, the package is only added to the list of packages, we could maybe also add it to the ‘default’ group.

Note that after loading the package, it is still necessary to do `FreeTypeFontProvider current updateEmbeddedFreeTypeFonts` to make the font available. I’m not sure whether it would be appropriate to include that in the class initialization method of PTermDejaVuSansMono. It’s also done by `#updateFreeType` in FreeTypeSettings class which is used in its `#startUp:` method.

A more compact representation of the font data is used in PTermDejaVuSansMono compared to what is used in [SourceCodeProRegular](https://github.com/pharo-project/pharo/blob/bbebd06a1cb15a882ebe6e27fae2c713286e9ffe/src/EmbeddedFreeType/SourceCodeProRegular.class.st) and [SourceSansProRegular](https://github.com/pharo-project/pharo/blob/bbebd06a1cb15a882ebe6e27fae2c713286e9ffe/src/EmbeddedFreeType/SourceSansProRegular.class.st). Those two classes each have a method `#fontContentsData` that uses a literal byte array with the contents of the ‘.ttf’ file. PTermDejaVuSansMono has a method `#fontContentsEncodedData` that uses a literal array of strings with the base64 encoding of the gzip-compressed ‘.ttf’ file. The method was generated through the following snippet, it uses the ‘gzip’ command because I couldn’t tell how to do ‘best’ compression with GZipWriteStream:

```smalltalk
gzipData := (LibC resultOfCommand: ('gzip --stdout --no-name --best ''{1}''' format: {
	((IceRepository repositoryNamed: 'PTerm') location / 'fonts' / 'DejaVuSansMono.ttf') fullName }))
		asByteArray.
PTermDejaVuSansMono class compile: (String streamContents: [ :stream |
	stream nextPutAll: 'fontContentsEncodedData'; cr; cr.
	stream tab; nextPutAll: '^ #('; cr.
	(gzipData base64Encoded groupsOf: 92)
		do: [ :group | stream tab; tab; nextPutAll: group printString ]
		separatedBy: [ stream cr ].
	stream nextPutAll: ')' ]).
```

lxsang reviewed (commented):
@Rinzwind  Thanks for the PR!

I'm little bit concerned about the use of base64 encoding in this case. We can embed directly the byte-array of the compressed file. This allows to reduce the code size (i might be wrong) and avoid an unnecessary decoding step. For example:

```smalltalk
gzipData := (LibC resultOfCommand: ('gzip --stdout --no-name --best ''{1}''' format: {
	((IceRepository repositoryNamed: 'PTerm') location / 'fonts' / 'DejaVuSansMono.ttf') fullName }))
		asByteArray.
PTermDejaVuSansMono class compile: (String streamContents: [ :stream |
	stream nextPutAll: 'fontContentsEncodedData'; cr; cr.
	stream tab; nextPutAll: '^ #('.
	stream nextPutAll: (' ' join: gzipData).
	stream nextPutAll: ')' ]).
```

Other than that, i found it is a good idea to have an embedded default font. I propose:

- [ ] Remove the use of base64 encoding in `PTerm-Fonts/PTermDejaVuSansMono.class.st` ?
- [ ] Load this package by default by integrating it to the group default
- [ ] Use this embedded font package instead of downloading it from github  as currently done.
- [ ] Optional, add a similar class for Unifont but only load it on user demand (via the menu)

Rinzwind commented:
I added the package ‘PTerm-Fonts’ to the ‘default’ group and changed TerminalEmulator to not download the font files anymore.

I looked into embedding Unifont as well, but I’m not sure I understand the license. The [page ‘GNU Unifont’ on unifoundry.com](http://unifoundry.com/unifont/index.html) says: “As of Unifont version 13.0.04, the fonts are dual-licensed under the SIL Open Font License (OFL) version 1.1 and the GNU GPL 2+ with the GNU font embedding exception.” The [file ‘LICENSE.txt’](http://unifoundry.com/LICENSE.txt) says: “The license for the compiled fonts is covered by the SIL Open Font License version 1.1 and by the above GPL terms with the GNU font embedding exception”. It’s not clear to me whether they mean one of the two licenses can be chosen, or whether the terms of both licenses apply. The GNU font embedding exception seems to only apply to documents.
